pdr_backend/pdr_dashboard/util/db.py

```diff
@@ -75,16 +77,13 @@ def _query_db(self, query: str, scalar=False) -> Union[List[dict], Exception]:
             dict: Query result.
         """
         try:
-            db = DuckDBDataStore(self.lake_dir, read_only=True)
-
+            # If scalar, fetch a single result
             if scalar:
-                result = db.query_scalar(query)
-            else:
-                df = db.query_data(query)
-                result = df.to_pandas()
-
-            db.duckdb_conn.close()
-            return result
+                result = duckdb.execute(query).fetchone()
+                return result[0] if result and len(result) == 1 else result
+            # For non-scalar, fetch as a dataframe and convert to dictionary
```

Comment:
I panicked here. Please remove this comment. This is a dataframe and not a dictionary